src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -241,6 +241,24 @@ str2index(const std::string& str, bool _inUserDomain)
   return bdf2index(str, _inUserDomain);
 }
 
+void
+XBUtilities::xrt_xbutil_version_cmp()
+{
+  boost::property_tree::ptree pt_xrt;
+  xrt_core::get_xrt_info(pt_xrt);
+  boost::property_tree::ptree empty_ptree;
+
+  std::string xrt_version = pt_xrt.get<std::string>("version", "N/A");
+  const boost::property_tree::ptree& available_drivers = pt_xrt.get_child("drivers", empty_ptree);
+  for(auto& drv : available_drivers) {
+    const boost::property_tree::ptree& driver = drv.second;
+    std::string drv_name = driver.get<std::string>("name", "N/A");
+    std::string drv_version = driver.get<std::string>("version", "N/A");
+    if (drv_name.compare("xocl") == 0 && drv_version.compare("unknown") != 0 && xrt_version.compare(drv_version) != 0)
+      throw std::runtime_error("Mixed versions of XRT and xbutil are not supported. Please install matching versions of XRT and xbutil.");
```

Comment:
This solution does report the current build / version, but still doesn't help the user in understanding what is different.

It would be better the difference in versions was reported.   Something similar to but not limited to:

- Unknown driver version

- Unexpected XRT Driver version (2.13.0) found.  Was expected 2.14.0 which compatible with this version of XRT tools.

Reply:
Hi Stephen,

Thank you for the feedback; I have updated the error messages! However, while doing so, I realized that drivers with unknown versions were being let through without throwing an error. Do we want to continue allowing this? When throwing an error for unknown driver version, pipeline builds ended up failing due to the issue, though I am not entirely sure if this is expected behavior.